conf/machine/imx6qdlsabreauto.conf

```diff
@@ -40,22 +40,29 @@ UBOOT_SUFFIX:pn-u-boot-fslc = "img"
 ### u-boot-imx settings ###
 SPL_BINARY:pn-u-boot-imx = ""
 UBOOT_CONFIG:pn-u-boot-imx ??= " \
-    sd-imx \
-    ${@bb.utils.contains('MACHINE_FEATURES', 'optee', 'sd-optee-imx', '', d)} \
+    sd \
+    ${@bb.utils.contains('MACHINE_FEATURES', 'optee', 'sd-optee', '', d)} \
 "
 UBOOT_MAKE_TARGET:pn-u-boot-imx = "u-boot.imx"
 UBOOT_SUFFIX:pn-u-boot-imx = "imx"
 
 # The u-boot-imx does not provide unified functionality for DL/Q/QP SoC
 # variants. Change the defconfig to the targeted SoC variant.
-UBOOT_CONFIG[sd-fslc]      = "mx6sabreauto_defconfig,sdcard"
-UBOOT_CONFIG[sd-imx]       = "mx6qpsabreauto_defconfig,sdcard"
-UBOOT_CONFIG[sd-optee-imx] = "mx6qpsabreauto_optee_defconfig,sdcard"
-UBOOT_CONFIG[eimnor-imx]   = "mx6qpsabreauto_eimnor_defconfig"
-UBOOT_CONFIG[nand-imx]     = "mx6qpsabreauto_nand_defconfig,ubifs"
-UBOOT_CONFIG[spinor-imx]   = "mx6qpsabreauto_spinor_defconfig"
-UBOOT_CONFIG[sata-imx]     = "mx6qpsabreauto_sata_defconfig"
-UBOOT_CONFIG[mfgtool-imx]  = "mx6qpsabreauto_defconfig"
+UBOOT_CONFIG[sd]       = "${UBOOT_CONFIG_SD}"
+UBOOT_CONFIG[sd-optee] = "${UBOOT_CONFIG_SD_OPTEE}"
+UBOOT_CONFIG[eimnor]   = "${UBOOT_CONFIG_EIMNOR}"
+UBOOT_CONFIG[nand]     = "${UBOOT_CONFIG_NAND}"
+UBOOT_CONFIG[spinor]   = "${UBOOT_CONFIG_SPINOR}"
+UBOOT_CONFIG[sata]     = "${UBOOT_CONFIG_SATA}"
+UBOOT_CONFIG[mfgtool]  = "${UBOOT_CONFIG_MFGTOOL}"
+UBOOT_CONFIG_SD                   = "mx6sabreauto_defconfig,sdcard"
+UBOOT_CONFIG_SD:imx-nxp-bsp       = "mx6qpsabreauto_defconfig,sdcard"
+UBOOT_CONFIG_SD_OPTEE:imx-nxp-bsp = "mx6qpsabreauto_optee_defconfig,sdcard"
+UBOOT_CONFIG_EIMNOR:imx-nxp-bsp   = "mx6qpsabreauto_eimnor_defconfig"
+UBOOT_CONFIG_NAND:imx-nxp-bsp     = "mx6qpsabreauto_nand_defconfig,ubifs"
+UBOOT_CONFIG_SPINOR:imx-nxp-bsp   = "mx6qpsabreauto_spinor_defconfig"
+UBOOT_CONFIG_SATA:imx-nxp-bsp     = "mx6qpsabreauto_sata_defconfig"
+UBOOT_CONFIG_MFGTOOL:imx-nxp-bsp  = "mx6qpsabreauto_defconfig"
```

Comment:
What if you use a:

```conf
UBOOT_DEFCONFIG_NAME = "mx6sabreauto"
UBOOT_DEFCONFIG_NAME:pn-u-boot-imx = "mx6qpsabreauto"
```
So we can use this to expand the variable names?

We cannot rely on `imx-nxp-bsp` as we have users which uses the NXP BSP but using U-Boot mainline.

Reply:
Okay